src/improvements/single.just

```diff
@@ -73,35 +77,60 @@ sign hdPath='0':
 
   echo "Using script ${script}"
   echo "getting signer address..."
-  signer=$(cast wallet address --ledger --mnemonic-derivation-path "m/44'/60'/{{hdPath}}'/0/0")
-  echo "Signing with: ${signer}"
-  echo ""
-
-  # Reverts if signer is not an owner on the safe which executes this transaction.
-  forge script TaskManager --rpc-url ${rpcUrl} --sig "requireSignerOnSafe(address,string)" $signer $TASK_PATH
-
-  echo "⏳ Task signing in progress. Some tasks take longer than others..."
-  forge build
-  # Using the eip712sign within the repo folder since eip712sign was installed there in ./justfile.
-  $(git rev-parse --show-toplevel)/bin/eip712sign --ledger --hd-paths "m/44'/60'/{{hdPath}}'/0/0" -- \
-  forge script ${script} \
-    --rpc-url ${rpcUrl} \
-    --sig "simulateRun(string)" ${config}
+  if {{ledger}}; then
+    signer=$(cast wallet address --ledger --mnemonic-derivation-path "m/44'/60'/{{hdPath}}'/0/0")
+    echo "Signing with: ${signer}"
+    echo ""
+
+    # Reverts if signer is not an owner on the safe which executes this transaction.
+    forge script TaskManager --rpc-url ${rpcUrl} --sig "requireSignerOnSafe(address,string)" $signer $TASK_PATH
+
+    echo "⏳ Task signing in progress. Some tasks take longer than others..."
+    forge build
+    # Using the eip712sign within the repo folder since eip712sign was installed there in ./justfile.
+    $(git rev-parse --show-toplevel)/bin/eip712sign --ledger --hd-paths "m/44'/60'/{{hdPath}}'/0/0" -- \
+    forge script ${script} \
+      --rpc-url ${rpcUrl} \
+      --sig "simulateRun(string)" ${config}
+  else
+    cd ${TASK_PATH}
```

Comment:
why do we need this in this branch but not the other?

Reply:
What I'm confused about though is: 
1. We only do this `cd` when in the `else` case, but not the `if {{ledger}}` case.
2. I don't see it in the previous code for this diff. Maybe the `config=${TASK_PATH}/config.toml` on line 70 above is handling it already?

Reply:
Yes since there is a call to `just` for example you need to be in the correct folder: 

```
signer_private_key=$(just --justfile=${JUSTFILE_PATH} get-keystore-private-key)
```